Author: Sean Finney <seanius@debian.org>
Forwarded: no (upstream doesn't want it)
Description: Silence warnings about using the default system timezone info
 In vanilla upstream php, this is considered an error (i.e. the user must
 set the timezone explicitly), though with our use of the system timezonedb
 patch, we actually feel quite comfortable using the default timezone info.
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=571762
--- a/ext/date/php_date.c
+++ b/ext/date/php_date.c
@@ -878,7 +878,7 @@ static char* guess_timezone(const timeli
 			tzid = "UTC";
 		}
 		
-		php_error_docref(NULL TSRMLS_CC, E_WARNING, DATE_TZ_ERRMSG "We selected '%s' for '%s/%.1f/%s' instead", tzid, ta ? ta->tm_zone : "Unknown", ta ? (float) (ta->tm_gmtoff / 3600) : 0, ta ? (ta->tm_isdst ? "DST" : "no DST") : "Unknown");
+		// php_error_docref(NULL TSRMLS_CC, E_WARNING, DATE_TZ_ERRMSG "We selected '%s' for '%s/%.1f/%s' instead", tzid, ta ? ta->tm_zone : "Unknown", ta ? (float) (ta->tm_gmtoff / 3600) : 0, ta ? (ta->tm_isdst ? "DST" : "no DST") : "Unknown");
 		return tzid;
 	}
 #endif
--- a/ext/date/tests/date_default_timezone_get-1.phpt
+++ /dev/null
@@ -1,16 +0,0 @@
---TEST--
-date_default_timezone_get() function [1]
---INI--
-date.timezone=
---FILE--
-<?php
-	putenv('TZ=');
-	echo date_default_timezone_get(), "\n";
-	echo date('e'), "\n";
-?>
---EXPECTF--
-Warning: date_default_timezone_get(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected 'UTC' for 'UTC/0.0/no DST' instead in %sdate_default_timezone_get-1.php on line 3
-UTC
-
-Warning: date(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected 'UTC' for 'UTC/0.0/no DST' instead in %sdate_default_timezone_get-1.php on line 4
-UTC
--- a/ext/date/tests/date_default_timezone_get-2.phpt
+++ /dev/null
@@ -1,12 +0,0 @@
---TEST--
-date_default_timezone_get() function [2]
---INI--
-date.timezone=
---FILE--
-<?php
-	putenv('TZ=');
-	echo date_default_timezone_get(), "\n";
-?>
---EXPECTF--
-Warning: date_default_timezone_get(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected 'UTC' for 'UTC/0.0/no DST' instead in %sdate_default_timezone_get-2.php on line 3
-UTC
--- a/ext/date/tests/date_default_timezone_set-1.phpt
+++ b/ext/date/tests/date_default_timezone_set-1.phpt
@@ -18,9 +18,6 @@ date.timezone=
 	echo date(DATE_ISO8601, $date4), "\n";
 ?>
 --EXPECTF--
-Warning: strtotime(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected 'UTC' for 'UTC/0.0/no DST' instead in %sdate_default_timezone_set-1.php on line 3
-
-Warning: strtotime(): It is not safe to rely on the system's timezone settings. You are *required* to use the date.timezone setting or the date_default_timezone_set() function. In case you used any of those methods and you are still getting this warning, you most likely misspelled the timezone identifier. We selected 'UTC' for 'UTC/0.0/no DST' instead in %sdate_default_timezone_set-1.php on line 4
 America/Indiana/Knox
 2005-01-12T03:00:00-0500
 2005-07-12T03:00:00-0500
